client/src/lib/data.ts

```diff
@@ -60,3 +81,27 @@ export async function addEvent(newEvent: Partial<TripEntry>, iconUrl: string) {
     console.error('Fetch Error', error);
   }
 }
+
+export async function updateTrip(
+  editEvent: Partial<TripEntry>,
+  userId: number,
+  tripId: number,
```

Comment:
Isn't `editEvent` guaranteed to have a tripId in it? So you don't need to pass it in?

The way I like to do this is to just take the userId and `trip: TripEntry` and force the caller to update the iconUrl, etc. So I can pass any trip in I want and be confident that the database will be updated to match.